### PR TITLE
DE44122: FACE Weblinks > No error message shown when saving link that is missing http/https

### PR DESCRIPTION
**Pull request by coffeexcode**
Rally: https://rally1.rallydev.com/#/289692574792d/custom/355050439968?detail=%2Fdefect%2F602348528019

This PR changes the URL validation to require the input to have either http:// or https:// at the beginning of the URL.

Before:
There would be no indication of an error, and when trying to save nothing would occur. Network tab would show a failure.
image

After:
The user is notified that the URL is invalid before trying to submit.
image

## TODO
- [x] Another developer will devtest this on their machine

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-content-editor/web-link/helpers/url-validation-helper.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ export const getWeblinkError = (link, isExternalResource, isSubmitting = false)
 		return isSubmitting ? 'content.emptyLinkField' : null;
 	}
 
-	const expression = /^(?:https?:\/\/)?(?:[a-zA-Z0-9][a-zA-Z0-9-]*\.)+[a-zA-Z0-9][a-zA-Z0-9-]*(?::\d+)?(?:$|[/?#].*$)/;
+	const expression = /^(?:https?:\/\/){1}(?:[a-zA-Z0-9][a-zA-Z0-9-]*\.)+[a-zA-Z0-9][a-zA-Z0-9-]*(?::\d+)?(?:$|[/?#].*$)/;
```
Comment (ianlaidlaw) [inline components/d2l-activity-editor/d2l-activity-content-editor/web-link/helpers/url-validation-helper.js:7]: this regexp was taken directly from lessons, i wonder if we should make the same change there. the old view allows without http/https

**Review by ianlaidlaw — APPROVED** [review]
LGTM

**Review by quinnbudan — APPROVED** [review]
LGTM! Nice

**Review by brissons — APPROVED** [review]
lgtm! Pulled in the changes and it looks like it is behaving properly now :)

File: test/d2l-activity-editor/d2l-activity-content-editor/web-link/helpers/url-validation-helper.spec.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ describe('UrlValidationHelper', () => {
 
 	// TODO: Finish writing tests for this
```
Comment (CodeBaboon) [inline test/d2l-activity-editor/d2l-activity-content-editor/web-link/helpers/url-validation-helper.spec.js:8]: Is this TODO covered off now?

File: test/d2l-activity-editor/d2l-activity-content-editor/web-link/helpers/url-validation-helper.spec.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ describe('UrlValidationHelper', () => {
...
 	describe('getWeblinkError', () => {
-		describe('when url is embedded', () => {
+		describe('when url is not embedded', () => {
```
Comment (CodeBaboon) [inline test/d2l-activity-editor/d2l-activity-content-editor/web-link/helpers/url-validation-helper.spec.js:10]: nice catch here.

**Review by CodeBaboon — APPROVED** [review]

**Review by DiljotSG — APPROVED** [review]

**Pull request merged**